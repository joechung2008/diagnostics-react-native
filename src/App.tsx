import AppContent from './AppContent';
import { ThemeProvider } from './ThemeContext';
import { SafeAreaProvider } from 'react-native-safe-area-context';

function App() {
  return (
    <SafeAreaProvider>
      <ThemeProvider>
        <AppContent />
      </ThemeProvider>
    </SafeAreaProvider>
  );
}

export default App;
