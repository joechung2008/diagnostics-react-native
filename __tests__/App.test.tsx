import App from '@/App';
import { render } from '@testing-library/react-native';
import React from 'react';

// Mock child components
jest.mock('@/AppContent', () => 'AppContent');
jest.mock('@/ThemeContext', () => ({
  ThemeProvider: ({ children }: { children: React.ReactNode }) => (
    <>{children}</>
  ),
}));

describe('App', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<App />);
    expect(toJSON()).toMatchSnapshot();
  });
});
