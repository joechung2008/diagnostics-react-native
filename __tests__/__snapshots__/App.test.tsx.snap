// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`App renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
